src/libraries/System.Private.Runtime.InteropServices.JavaScript/tests/System/Runtime/InteropServices/JavaScript/MarshalTests.cs

```diff
@@ -1019,5 +1022,193 @@ public static async Task MarshalFailedAsynchronousValueTaskDoesNotWorkYet()
+            );
+            Assert.Equal(ExpectedDateTime, HelperMarshal._dateTimeValue);
+        }
+
+        [Fact]
+        [ActiveIssue("https://github.com/dotnet/runtime/pull/47640")]
+        public static void ReturnCustomClass()
+        {
+            HelperMarshal._ccValue = new HelperMarshal.CustomClass ();
+            Runtime.InvokeJS(
+                "var cc = App.call_test_method ('ReturnCustomClass', [ 4.13 ], 'a');" +
+                "App.call_test_method ('InvokeCustomClass', [ cc ], 'a');"
+            );
+            Assert.Equal(4.13, HelperMarshal._ccValue?.D);
+        }
+
+        [Fact]
+        [ActiveIssue("https://github.com/dotnet/runtime/pull/47640")]
+        public static void ReturnCustomStruct()
+        {
+            HelperMarshal._csValue = default(HelperMarshal.CustomStruct);
+            Runtime.InvokeJS(
+                "var cs = App.call_test_method ('ReturnCustomStruct', [ 4.13 ], 'a');" +
+                "App.call_test_method ('InvokeCustomStruct', [ cs ], 'a');"
+            );
+            Assert.Equal(4.13, HelperMarshal._csValue.D);
+        }
+
+        [Fact]
+        [ActiveIssue("https://github.com/dotnet/runtime/pull/47640")]
+        public static void ReturnCustomDate()
+        {
+            HelperMarshal._dateTimeValue = default(DateTime);
+            Runtime.InvokeJS(
+                $"var dt = new Date('{ExpectedDateString}');\r\n" +
+                "var cd = App.call_test_method ('ReturnCustomDate', [ dt ], 'a');" +
+                "App.call_test_method ('InvokeCustomDate', [ cd ], 'a');"
+            );
+            Assert.Equal(ExpectedDateTime, HelperMarshal._dateTimeValue);
+        }
+
+        [Fact]
+        [ActiveIssue("https://github.com/dotnet/runtime/pull/47640")]
+        public static void InvokeCustomVector3()
+        {
+            HelperMarshal._vec3Value = default(HelperMarshal.CustomVector3);
+            Runtime.InvokeJS(
+                "App.call_test_method ('InvokeCustomVector3', [ [1, 2.5, 4] ], 'a');"
+            );
+            Assert.Equal(1, HelperMarshal._vec3Value.X);
+            Assert.Equal(2.5, HelperMarshal._vec3Value.Y);
+            Assert.Equal(4, HelperMarshal._vec3Value.Z);
+        }
+
+        [Fact]
+        [ActiveIssue("https://github.com/dotnet/runtime/pull/47640")]
+        public static void ReturnCustomVector3()
+        {
+            HelperMarshal._vec3Value = default(HelperMarshal.CustomVector3);
+            Runtime.InvokeJS(
+                "var cv3 = App.call_test_method ('ReturnCustomVector3', [ [1, 2.5, 4] ], 'a');" +
+                "App.call_test_method ('InvokeCustomVector3', [ cv3 ], 'a');"
+            );
+            Assert.Equal(1, HelperMarshal._vec3Value.X);
+            Assert.Equal(2.5, HelperMarshal._vec3Value.Y);
+            Assert.Equal(4, HelperMarshal._vec3Value.Z);
+        }
+
+        [Fact]
+        [ActiveIssue("https://github.com/dotnet/runtime/pull/47640")]
```

Comment:
Could you please create issue(s) instead of PR for all `[ActiveIssue]`. 
Ideally with good description of missing functionality. 
Multiple issues if the underlying concepts are separate. 
Many thanks!